pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -189,16 +189,17 @@ func (r Requirements) Validate() (errs error) {
 // Compatible ensures the provided requirements can be met.
 func (r Requirements) Compatible(requirements Requirements) (errs error) {
 	for key, requirement := range requirements.requirements {
-		if requirement.Type() == v1.NodeSelectorOpIn && requirement.Intersection(r.Get(key)).Len() == 0 {
+		intersection := requirement.Intersection(r.Get(key))
+		if requirement.Type() == v1.NodeSelectorOpIn && intersection.Len() == 0 {
 			errs = multierr.Append(errs, fmt.Errorf("%s not in %s, key %s", requirement, r.Get(key), key))
 		}
-		if requirement.Type() == v1.NodeSelectorOpExists && requirement.Intersection(r.Get(key)).Len() == 0 {
+		if requirement.Type() == v1.NodeSelectorOpExists && intersection.Len() == 0 {
 			errs = multierr.Append(errs, fmt.Errorf("%s not in %s, key %s", requirement, r.Get(key), key))
 		}
-		if requirement.Type() == v1.NodeSelectorOpNotIn && r.Get(key).Type() == v1.NodeSelectorOpIn && requirement.Intersection(r.Get(key)).Len() == 0 {
+		if requirement.Type() == v1.NodeSelectorOpNotIn && r.Get(key).Type() == v1.NodeSelectorOpIn && intersection.Len() == 0 {
 			errs = multierr.Append(errs, fmt.Errorf("%s not in %s, key %s", requirement, r.Get(key), key))
 		}
-		if requirement.Type() == v1.NodeSelectorOpDoesNotExist && r.Get(key).Len() > 0 {
+		if requirement.Type() == v1.NodeSelectorOpDoesNotExist && r.Get(key).Type() == v1.NodeSelectorOpIn {
```

Comment:
I think we should also raise error for r.Get(key).Type() == v1.NodeSelectorOpExists